triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/TriggerProcess.scala

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger.simulation
+package process
+
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.actor.typed.scaladsl.Behaviors
+import akka.stream.Materializer
+import akka.util.Timeout
+import com.daml.ledger.api.refinements.ApiTypes.{ApplicationId, Party}
+import com.daml.ledger.api.v1.event.CreatedEvent
+import com.daml.ledger.api.v1.transaction_filter.TransactionFilter
+import com.daml.ledger.client.LedgerClient
+import com.daml.lf.PureCompiledPackages
+import com.daml.lf.data.Ref.{Identifier, PackageId, QualifiedName}
+import com.daml.lf.engine.trigger.simulation.TriggerMultiProcessSimulation.{
+  TriggerSimulationConfig,
+  TriggerSimulationFailure,
+}
+import com.daml.lf.engine.trigger.{
+  Converter,
+  Runner,
+  TriggerMsg,
+  TriggerParties,
+  TriggerRunnerConfig,
+}
+import com.daml.lf.speedy.SValue
+import com.daml.platform.services.time.TimeProviderType
+
+import java.lang.management.ManagementFactory
+import java.util.UUID
+import scala.collection.immutable.TreeMap
+import scala.concurrent.Await
+import scala.util.{Failure, Success}
+import scalaz.syntax.tag._
+
+import scala.jdk.CollectionConverters._
+
+object TriggerProcess {
```

Comment:
The trigger process wraps up the `updateState` lambda for each Daml trigger with a Speedy evaluation layer of code